test/base/src/androidTest/kotlin/io/realm/test/shared/QueryTests.kt

```diff
@@ -154,35 +157,59 @@ class QueryTests {
     fun find_mutableRealm_malformedQueryThrows() {
         realm.writeBlocking {
             assertFailsWith<IllegalArgumentException> {
-                query(QuerySample::class, "stringField = $0")
+                query<QuerySample>("stringField = $0")
             }.let {
                 assertTrue(it.message!!.contains("Have you specified all parameters"))
             }
 
             assertFailsWith<IllegalArgumentException> {
-                query(QuerySample::class, "stringField = 42")
+                query<QuerySample>("stringField = 42")
             }.let {
                 assertTrue(it.message!!.contains("Wrong query field"))
             }
 
             assertFailsWith<IllegalArgumentException> {
-                query(QuerySample::class, "nonExistingField = 13")
+                query<QuerySample>("nonExistingField = 13")
             }.let {
                 assertTrue(it.message!!.contains("Wrong query field"))
             }
         }
     }
 
     @Test
-    @Ignore
     fun asFlow() {
```

Comment:
Aren't there more corner cases that we should test? Empty results, deleting parent object, etc.?

Reply:
Empty results are being tested here, but I will split the test in two for better visibility. I forgot about deleting the observable object 🙈